store.go

```diff
@@ -647,17 +647,23 @@ func GetStore(options types.StoreOptions) (Store, error) {
 	storesLock.Lock()
 	defer storesLock.Unlock()
 
+	// return if BOTH run and graph root are matched, otherwise our run-root can be overriden if the graph is found first
 	for _, s := range stores {
-		if s.graphRoot == options.GraphRoot && (options.GraphDriverName == "" || s.graphDriverName == options.GraphDriverName) {
+		if (s.graphRoot == options.GraphRoot) && (s.runRoot == options.RunRoot) && (options.GraphDriverName == "" || s.graphDriverName == options.GraphDriverName) {
 			return s, nil
 		}
 	}
 
-	if options.GraphRoot == "" {
-		return nil, errors.Wrap(ErrIncompleteOptions, "no storage root specified")
-	}
-	if options.RunRoot == "" {
-		return nil, errors.Wrap(ErrIncompleteOptions, "no storage runroot specified")
+	// if passed a run-root or graph-root alone, the other should be defaulted only error if we have neither.
+	switch {
+	case options.RunRoot != "" && options.GraphRoot != "":
+		break
+	case options.GraphRoot == "" && options.RunRoot != "":
```

Comment:
No need for && I think this is simpler to read as 
```
case options.Graphroot == "":
                options.GraphRoot = types.Options().GraphRoot
case options.RunRoot == "":
		options.RunRoot = types.Options().RunRoot
```

Reply:
ok, I'm going to put the `case options.RunRoot == "" && options.GraphRoot == ""` at the top then since I think we want that to return an error, we need either a graph or a run dir to be passed or set by this point.